rectools/models/nn/transformer_base.py

```diff
@@ -294,20 +301,84 @@ def _init_trainer(self) -> None:
         else:
             self._trainer = self.get_trainer_func()
 
-    def _init_torch_model(self) -> TransformerTorchBackbone:
-        return TransformerTorchBackbone(
+    def init_item_net_from_dataset(self, dataset: Dataset) -> ItemNetConstructorBase:
+        """
+        Construct network for item embeddings from dataset.
+
+        Parameters
+        ----------
+        dataset : Dataset
+            RecTools dataset with user-item interactions.
+
+        Returns
+        -------
+        ItemNetConstructorBase
+            Network for item embeddings.
+        """
+        return self.item_net_constructor_type.from_dataset(
+            dataset, self.n_factors, self.dropout_rate, self.item_net_block_types
+        )
+
+    def init_item_net_from_dataset_schema(self, dataset_schema: DatasetSchema) -> ItemNetConstructorBase:
+        """
+        Construct network for item embeddings from dataset schema.
+
+        Parameters
+        ----------
+        dataset_schema : DatasetSchema
+            RecTools schema with dataset statistics.
+
+        Returns
+        -------
+        ItemNetConstructorBase
+            Network for item embeddings.
+        """
+        return self.item_net_constructor_type.from_dataset_schema(
+            dataset_schema, self.n_factors, self.dropout_rate, self.item_net_block_types
+        )
+
+    def init_pos_encoding_layer(self) -> PositionalEncodingBase:
+        """
+        Construct positional encoding layer.
+
+        Returns
+        -------
+        PositionalEncodingBase
+            Positional encoding layer.
+        """
+        return self.pos_encoding_type(self.use_pos_emb, self.session_max_len, self.n_factors)
+
+    def init_transformer_layers(self) -> TransformerLayersBase:
+        """
+        Construct transformer layers.
+
+        Returns
+        -------
+        TransformerLayersBase
+            Transformer layers.
+        """
+        return self.transformer_layers_type(
             n_blocks=self.n_blocks,
             n_factors=self.n_factors,
             n_heads=self.n_heads,
-            session_max_len=self.session_max_len,
             dropout_rate=self.dropout_rate,
-            use_pos_emb=self.use_pos_emb,
+        )
+
+    def _init_torch_model(self, dataset_schema: tp.Optional[DatasetSchema] = None) -> TransformerTorchBackbone:
+        if dataset_schema is None:
```

Comment:
This is not very clear.
Let's construct item net first. And then pass it to `_init_torch_model` method.

Reply:
fixed